src/background/dev-tools-monitor.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { OnDevToolStatusPayload } from 'background/actions/action-payloads';
+import { TabContextManager } from 'background/tab-context-manager';
+import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
+import { Messages } from 'common/messages';
+import { PromiseFactory, TimeoutError } from 'common/promises/promise-factory';
+import { DevToolsStatusResponse } from 'common/types/dev-tools-messages';
+import _ from 'lodash';
+
+export class DevToolsMonitor {
+    constructor(
+        private readonly browserAdapter: BrowserAdapter,
+        private readonly promiseFactory: PromiseFactory,
+        protected readonly activeDevtoolTabIds: number[],
+        private readonly tabContextManager: TabContextManager,
+        private readonly messageTimeoutMilliseconds = 5000,
```

Comment:
The PR description says that the timeout will be 500 ms but this has an extra 0 on the end- which is the correct timeout?

Reply:
My mistake, I changed some of the timeouts while I was testing and forgot to put them back